src/model_instance_state.cu

```diff
@@ -33,54 +34,86 @@ ModelInstanceState::Create(
     ModelState* model_state, TRITONBACKEND_ModelInstance* triton_model_instance)
 {
   std::string instance_name = get_model_instance_name(*triton_model_instance);
-  TRITONSERVER_InstanceGroupKind instance_kind =
-      get_instance_kind(*triton_model_instance);
   int32_t instance_id = get_device_id(*triton_model_instance);
 
   return std::make_unique<ModelInstanceState>(
-      model_state, triton_model_instance, instance_name.c_str(), instance_kind,
-      instance_id);
+      model_state, triton_model_instance, instance_name.c_str(), instance_id);
 }
 
-raft::handle_t&
+raft::handle_t*
```

Comment:
Since we're not inherently doing address manipulation or pointer arithmetic with this (the only conditions for which C++ core guidelines suggest using a raw pointer), it would probably be better to use an `optional` instead of returning a raw pointer, especially since we have `optional` implemented for pre-C++17 [here](https://github.com/triton-inference-server/fil_backend/blob/main/src/triton_fil/optional.h).

Reply:
It is actually not possible to use `optional<raft::handle_t>`, since `raft::handle_t` is not copy assignable or constructible.

Reply:
Now that we're targeting 21.07, it probably makes sense to take care of this in this PR rather than deferring. As we talked about offline, `optional` should be viable, possibly combined with `std::ref` if necessary.